doc/developer/design/20240411_graceful_reconfiguration_of_managed_clusters_v1.md

```diff
@@ -0,0 +1,329 @@
+an alter cluster on a managed cluster.
+````
+NOTE:
+This assumes we're storing prior config in the reconfig schedule, if we're
+not we can probably just compare against the current config in which case we
+wouldn't allow any cancelation.
+````
+
+In sequence_alter_cluster_managed_to_managed:
+ Is there a reconfiguration going on.. 
+   - no
+      - if delay = false
+        - use current logic
+      - if dealy = true
+        - create new replicas (but do not drop the old ones).
+        - set the cluster reconfiugration scheulde to specified duraiton (defaults to 0)
+   - yes
+     - if delay = true
+       - if new config = prior config (ex: new size = size stored in `ClusterScheudle::Reconfigure.prior_config`)
+         -  The user is effectively cancelling an in-progress
+            reconfiguration, drop the reconf replicas. (And possibly adjust
+            the number of stable replicas if the replication factor has changed.)
+            reset the cluster schedule.
+       - if config = current_config (ex: new size = current catalog size)
+         - adjust the number of reconf replicas if the replication factor changed and
+           adjust the reconfigure schedule deadline if it has changed.
+          // POTENTIAL ISSUE: Using reconfiguration deadline seems like it
+          //  may run into some issues. If the same command is run twice do
+          //  we just bump up the deadline? It does have the nice behavior
+          //  where you could set the timeout to 0 and force an early
+          //  reoncifigure_finalization.
+       - if new config != prior or current
+         -  drop the reconf replicas, and create new reconf replicas.
+            update the reconfiguration deadline as needed
+     - if delay = false
+       - follow the old logic of dropping and re-creating, drop all replicas,
+         including reconfig replcias, reset the cluster schedule.
+```
+CONCERN:
+Allowing delay=false and a timeout=0 provides two similar mechanisms of
+using the old behavior. I think this is what we want as it ensures the old
+replicas are destroyed before the new ones are created which has billining
+implications. If we were to always rely on the reconfiguration_finalization
+we may charge for both sets of replicas for some additional time if the
+coordinator has a stall. If we're not concerned about this, always using
+reconfiguration_finalization may be better as it reduces the code paths.
+```
+
+## Open Questions
+
+### How do we handle multiple schedules?
+If this feature is introducing multiple schedules we'll need some rules governing the interaction between schedules.
+There's world where each schedule needs to consider it's interaction with all other schedules and that seems kind of
+scary.
+
+### How does a multi-subscriber coord work with schedules?
+We definitely don't want two schedule decisions to be acting at the same time. Currently this isn't
+an issue, but it seems a bit scary. Is this a concern?
```

Comment:
I'd guess that stuff like altering clusters will always be a centralized thing, and only 1 instance of  `handle_scheduling_decisions` would be allowed to run at a time. (But I'm not an expert on PlatformV2 stuff.)